src/Datadog.Trace/Tagging/MsmqTags.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.ExtensionMethods;
+
+namespace Datadog.Trace.Tagging
+{
+    internal class MsmqTags : InstrumentationTags
+    {
+        protected static readonly IProperty<string>[] MsmqTagsProperties =
+           InstrumentationTagsProperties.Concat(
+               new Property<MsmqTags, string>(Trace.Tags.InstrumentationName, t => t.InstrumentationName, (t, v) => t.InstrumentationName = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqCommand, t => t.Command, (t, v) => t.Command = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqQueue, t => t.Queue, (t, v) => t.Queue = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqQueueLabel, t => t.QueueLabel, (t, v) => t.QueueLabel = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqQueueLastModifiedTime, t => t.QueueLastModifiedTime, (t, v) => t.QueueLastModifiedTime = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqQueueUniqueName, t => t.UniqueQueueName, (t, v) => t.UniqueQueueName = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqIsTransactionalQueue, t => t.IsTransactionalQueue, (t, v) => t.IsTransactionalQueue = v),
+               new Property<MsmqTags, string>(Trace.Tags.MsmqTransactionType, t => t.TransactionType, (t, v) => t.TransactionType = v));
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MsmqTags"/> class.
+        /// </summary>
+        /// <param name="spanKind">kind of span</param>
+        public MsmqTags(string spanKind) => SpanKind = spanKind;
```

Comment:
You'll need to create a default constructor version of this too, as there's a unit tests that checks for its existence